src/rust/engine/options/src/args_tests.rs

```diff
@@ -194,29 +202,35 @@ fn test_scalar_fromfile() {
     fn do_test<T: PartialEq + Debug>(
         content: &str,
         expected: T,
-        getter: fn(&Args, &OptionId) -> Result<Option<T>, String>,
+        getter: fn(&ArgsReader, &OptionId) -> Result<Option<T>, String>,
         negate: bool,
     ) {
         let (_tmpdir, fromfile_path) = write_fromfile("fromfile.txt", content);
-        let args = Args::new(vec![format!(
+        let args = mk_args(vec![format!(
```

Comment:
Here and below we should have used `mk_args` (suitably generalized, as above) all along. Now it's more succinct and so more important to do so.